olive/data_config/config.py

```diff
@@ -0,0 +1,193 @@
+            params = signature(component).parameters
+            for param, info in params.items():
+                # 2. override the params with params_config
+                if param in self.params_config:
+                    v.params[param] = self.params_config[param]
+                    continue
+                # 3. if it already defined params under the component, use the params directly
+                if param not in v.params and not param.startswith("_"):
+                    if info.kind == info.VAR_POSITIONAL or info.kind == info.VAR_KEYWORD:
+                        continue
+                    elif info.default is info.empty:
+                        logger.debug(f"Missing parameter {param} for component {k}")
+                        v.params[param] = None
+                    else:
+                        v.params[param] = params[param].default
+
+    def get_components_params(self):
+        """
+        Get the parameters of the data container.
+        """
+        return {k: v.params for k, v in self.components.items()}
+
+    @property
+    def dataset(self):
+        """
+        Get the dataset of the data container.
+        """
+        name = self.components["dataset"].type or DefaultDataComponent.DATASET.value
+        return Registry.get_dataset_component(name)
+
+    @property
+    def pre_process(self):
+        """
+        Get the pre-process of the data container.
+        """
+        name = self.components["pre_process"].type or DefaultDataComponent.PRE_PROCESS.value
+        return Registry.get_pre_process_component(name)
+
+    @property
+    def post_process(self):
+        """
+        Get the post-process of the data container.
+        """
+        name = self.components["post_process"].type or DefaultDataComponent.POST_PROCESS.value
+        return Registry.get_post_process_component(name)
+
+    @property
+    def dataloader(self):
+        """
+        Get the dataloader of the data container.
+        """
+        name = self.components["dataloader"].type or DefaultDataComponent.DATALOADER.value
+        return Registry.get_dataloader_component(name)
+
+    @property
+    def dataset_params(self):
+        """
+        Get the parameters of the dataset.
+        """
+        return self.components["dataset"].params
+
+    @property
+    def pre_process_params(self):
+        """
+        Get the parameters of the pre-process.
+        """
+        return self.components["pre_process"].params
+
+    @property
+    def post_process_params(self):
+        """
+        Get the parameters of the post-process.
+        """
+        return self.components["post_process"].params
+
+    @property
+    def dataloader_params(self):
+        """
+        Get the parameters of the dataloader.
+        """
+        return self.components["dataloader"].params
+
+    def to_data_container(self):
+        """
+        Convert the data container config to the data container.
```

Comment:
>Convert the data container config to the data container [](http://example.com/codeflow?start=8&length=55)

Convert the data configuration to Olive dataset.

Reply:
Ditto.